test/e2e/storage/csi_volumes.go

```diff
@@ -0,0 +1,544 @@
+	ns string,
+) error {
+	volumeAttachementClient := client.StorageV1alpha1().VolumeAttachments()
+	return wait.Poll(2*time.Second, 10*time.Minute, func() (bool, error) {
+		va, err := volumeAttachementClient.Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return va.Status.Attached, nil
+	})
+}
+
+var _ = SIGDescribe("CSI Volumes [Feature:CSI]", func() {
+	f := framework.NewDefaultFramework("csi-mock-plugin")
+
+	var (
+		cs     clientset.Interface
+		ns     *v1.Namespace
+		node   v1.Node
+		config framework.VolumeTestConfig
+		suffix string
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		node = nodes.Items[rand.Intn(len(nodes.Items))]
+		config = framework.VolumeTestConfig{
+			Namespace:         ns.Name,
+			Prefix:            "csi",
+			ClientNodeName:    node.Name,
+			WaitForCompletion: true,
+		}
+		suffix = ns.Name
+	})
+
+	Describe("external attacher and mock plugin only", func() {
+
+		var (
+			clusterRole    *rbacv1.ClusterRole
+			serviceAccount *v1.ServiceAccount
+			pod            *v1.Pod
+		)
+
+		const (
+			mockPluginVolumeHandle = "1"
+			mockPluginNodeID       = "csi-mock"
+			mockPluginName         = "csi-mock"
+		)
+
+		BeforeEach(func() {
+			By("cleaning up any left over objects")
+			externalAttacherVolumeAttachment(cs, config, true, true, node.GetName(), mockPluginName, csiPvName)
+			externalAttacherPersistentVolume(cs, config, true, true, mockPluginName, mockPluginVolumeHandle)
+
+			By("deploying the external attacher")
+			clusterRole = externalAttacherClusterRole(cs, config, false)
+			serviceAccount = externalAttacherServiceAccout(cs, config, false)
+			externalAttacherClusterRoleBinding(cs, config, false, serviceAccount, clusterRole)
+			pod = externalAttacherPodWithMockPlugin(cs, config, false, f, serviceAccount)
+		})
+
+		AfterEach(func() {
+			By("uninstalling external attacher")
+			pod = externalAttacherPodWithMockPlugin(cs, config, true, f, serviceAccount)
+			externalAttacherClusterRoleBinding(cs, config, true, serviceAccount, clusterRole)
+			serviceAccount = externalAttacherServiceAccout(cs, config, true)
+			clusterRole = externalAttacherClusterRole(cs, config, true)
+		})
+
+		It("should detect and update the VolumeAttacher object", func() {
+			By("setting node id on node")
+			setNodeAnnotation(cs, config, &node, mockPluginName, mockPluginNodeID)
+
+			By("verifying nothing is published")
+			output := f.ExecCommandInContainer(pod.GetName(), "mock-plugin",
+				"csc", "controller", "list-volumes", "-v", "0.1.0", "--endpoint", "unix://"+csiSocketPath)
+			Expect(strings.Contains(output, "csi-mock/dev")).ToNot(BeTrue())
+			Expect(strings.Contains(output, "csi-mock/mnt/mock")).ToNot(BeTrue())
+
+			By("creating a persistentvolume")
+			pv := externalAttacherPersistentVolume(cs, config, false, false, mockPluginName, mockPluginVolumeHandle)
+
+			By("creating a fake volume attacher")
```

Comment:
shouldn't the attach detach controller be the one creating/deleting the AttachVolume object?

Reply:
Currently that is not the task of the external-attacher. Instead is part of the plugin code created by #54529

Reply:
E2E tests are supposed to test from the user's POV.  If you're stubbing out different components, that is better suited for an integration or unit test.

Reply:
Yeah I agree, but this is the only way to test a ...mostly.. e2e due to csi components. I will add to this test the e2e for #54529

Reply:
Ok, I will add to this PR another commit for a full e2e test.